### ISSUE TITLE
Window on more than one monitor

### ISSUE THREAD

**Issue by prifre** [fyne-io/fyne#2809]
Currently window-support only exists and it is not possible to position windows, so no multiple-monitor support?
It would be nice it Fyne could check on available monitors so it would be possible to start on a second, third, etc monitor.

**Comment by Jacalz** [comment]
Hmm, I think this could be a hard one. We have been against positioning windows because it’s something that the window manager ideally should handle.

Opening on a different monitor might also fall into that category, I think. I would personally be very annoyed if I opened an application on my first monitor and it then opens on my third.

**Comment by andydotxyz** [comment]
For abitrary window positioning this would be not great. However there are a number of use-cases where a secondary screen is looking to be used, if that is the case then perhaps `ShowFullScreenOnAlternateMonitor` or similar?
Is this what you were describing @prifre?

**Comment by Bluebugs** [comment]
It would be interesting to figure an API that is meaningful still in the context of Wayland where it is not possible and X11 where the window manager can decide whatever it want.

**Comment by juztin** [comment]
On Manjaro, this has worked largely okay for me _(minus some weird widget sizing/placement when moving between monitors of different resolutions)_
Although, after the most recent update from Manjaro, Fyne apps now only load on my main monitor. They freeze when running on any non-main display.

**Comment by andydotxyz** [comment]
That's no good! Please open an issue so we can get that fixed :)

**Comment by Jacalz** [comment]
Also, please make sure to include what desktop environment your are using, within that issue. Manjaro offers a bunch of different configurations if I remover correctly

**Comment by J7mbo** [comment]
I understand from this thread that the window manager should handle positioning of windows.

So when I have a Mac with an external monitor and I write code to create a window that overflows onto the second monitor naturally (as handled by the window manager), why doesn't it work?

I thought I'd post this here as I believe it to be relevant to the conversation but if not I can create another issue.

Monitor 1 (1440 width), Monitor 2 (3440 width). Creating a window of width 2000 I would expect it to overflow from the first monitor onto the second because of the explanations given in this thread, but it doesn't do that.

**Comment by andydotxyz** [comment]
macOS doesn't allow that by default - it crops windows that overflow the screen edge. 
If you go to system preferences and in "mission control" turn off "displays have separate spaces" it should start working as you would expect.

**Comment by ndianabasi** [comment]
I've been exploring fyne and have a use case where I need to programmatically display frameless windows on secondary screens. As a matter of fact, the frameless windows MUST no be displayed on the primary screen. Example is use case is a presentation software.

From the look of things, this requirement was never prioritised. :(

**Comment by andydotxyz** [comment]
Yes, this is an area that I would like to see resolved as well. I think the complexity is to make an API that is clear but also cross-platform. The concept of a secondary display is very different in various scenarios.

i.e. is it "external display" or "monitor 2" or "presenting screen". I think we have to understand and support all of these whilst keeping things clean. I wonder if a const/enum-like expression of the monitor intent and a "Request Monitor" API or similar. We may also consider a "HasMonitor" API too so apps can do different things if the external is not present (i.e. if it's a slideshow you only want to show the presenter display if you have multiple screens).
